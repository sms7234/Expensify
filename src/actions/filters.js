export const setTagFilter = (tag='') => ({
  type: 'SET_TAG',
  tag
});

export const setAccountFilter = (account='') => ({
  type: 'SET_ACCOUNT',
  account
});

export const setCategoryFilter = (category='') => ({
  type: 'SET_CATEGORY',
  category
});

export const setBusinessFilter = (business='') => ({
  type: 'SET_BUSINESS',
  business
});

export const setNoteFilter = (note='') => ({
  type: 'SET_NOTE',
  note
});

export const sortByAmount = () => ({
  type: 'SET_SORTBY',
  sortBy: 'amount'
});

export const sortByDate = () => ({
  type: 'SET_SORTBY',
  sortBy: 'date'
});

export const setStartDate = (startDate) => ({
  type: 'SET_START-DATE',
  startDate
});

export const setEndDate = (endDate) => ({
  type: 'SET_END-DATE',
  endDate
});
