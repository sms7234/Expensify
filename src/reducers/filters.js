import moment from 'moment';

const filtersReducersDefaultState = {
  category: '',
  business: '',
  note: '',
  sortBy: 'date',
  startDate: moment().startOf('month'),
  endDate: moment().endOf('month')
};
export default (state=filtersReducersDefaultState,action) => {
  switch(action.type){
    case 'SET_CATEGORY':
      return { ...state, category: action.category}
    case 'SET_BUSINESS':
      return { ...state, business: action.business}
    case 'SET_NOTE':
      return { ...state, note: action.note}
    case 'SET_SORTBY':
      return {...state, sortBy: action.sortBy}
    case 'SET_START-DATE':
      return {...state, startDate: action.startDate}
      case 'SET_END-DATE':
        return {...state, endDate: action.endDate}
    default:
    return state;
  };
};
