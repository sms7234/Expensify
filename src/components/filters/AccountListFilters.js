import React from 'react';
import {connect} from 'react-redux';
import {setAccountFilter} from '../../actions/filters';

export class AccountListFilters extends React.Component {
  onTextChange = (e) => {
    this.props.setAccountFilter(e.target.value);
  };
  render () {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="search accounts"
              value={this.props.filters.account}
              onChange={this.onTextChange}
            />
          </div>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setAccountFilter: (text) => dispatch(setAccountFilter(text)),
});

export default connect(mapStateToProps, mapDispatchToProps)(AccountListFilters);
