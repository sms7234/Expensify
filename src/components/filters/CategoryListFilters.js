import React from 'react';
import {connect} from 'react-redux';
import {setCategoryFilter} from '../../actions/filters';

export class CategoryListFilters extends React.Component {
  onTextChange = (e) => {
    this.props.setCategoryFilter(e.target.value);
  };
  render () {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="search categories"
              value={this.props.filters.category}
              onChange={this.onTextChange}
            />
          </div>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setCategoryFilter: (text) => dispatch(setCategoryFilter(text)),
});

export default connect(mapStateToProps, mapDispatchToProps)(CategoryListFilters);
