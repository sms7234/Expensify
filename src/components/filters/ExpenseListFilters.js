import React from 'react';
import {connect} from 'react-redux';
import Accordion from 'react-bootstrap/Accordion';
import Card from 'react-bootstrap/Card';
import {setCategoryFilter, setBusinessFilter, setNoteFilter} from '../../actions/filters';
import {sortByAmount, sortByDate, setStartDate, setEndDate} from '../../actions/filters';
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onDatesChange = ({startDate, endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({calendarFocused}))
  };
  onSortChange = (e) => {
    if(e.target.value ==="amount") {
      this.props.sortByAmount();
    }else if (e.target.value ==="date") {
      this.props.sortByDate();
    }
  };
  onCategoryChange = (e) => {
    this.props.setCategoryFilter(e.target.value);
  };
  onBusinessChange = (e) => {
    this.props.setBusinessFilter(e.target.value);
  };
  onNoteChange = (e) => {
    this.props.setNoteFilter(e.target.value);
  };
  render() {
    return (
      <div className="content-container accordion__filters">
        <div className="input-group__item">
          <DateRangePicker
            startDate={this.props.filters.startDate}
            endDate={this.props.filters.endDate}
            startDateId="dasewqeqwe"
            endDateId="jijhjkyjyihi"
            onDatesChange={this.onDatesChange}
            focusedInput = {this.state.calendarFocused}
            onFocusChange = {this.onFocusChange}
            isOutsideRange = {() => false}
            numberOfMonths = {1}
            showClearDates={true}
          />
        </div>
      <Accordion>
        <Card>
          <Accordion.Toggle as={Card.Header} className="accordion__header" eventKey="0">
            Additional Filters
          </Accordion.Toggle>
          <Accordion.Collapse eventKey="0">
            <Card.Body>
              <div className="input-group">
                <div className="input-group--filters">
                  <div className="input-group__item">
                    <input type = "text"
                      className="text-input--filters"
                      placeholder="search categories"
                      value={this.props.filters.category} onChange={this.onCategoryChange}/>
                  </div>
                  <div className="input-group__item">
                    <input type = "text"
                      className="text-input--filters"
                      placeholder="search business"
                      value={this.props.filters.business} onChange={this.onBusinessChange}/>
                  </div>
                  <div>
                    <div className="input-group__item">
                      <input type = "text"
                        className="text-input--filters"
                        placeholder="search notes"
                        value={this.props.filters.note} onChange={this.onNoteChange}/>
                    </div>
                  </div>
                </div>
                <div>
                  <div className="input-group__item">
                    <select
                      className="select"
                      value={this.props.filters.sortBy}
                      onChange={this.onSortChange}>
                      <option value="date">Date</option>
                      <option value="amount">Amount</option>
                    </select>
                  </div>
                </div>
              </div>
            </Card.Body>
          </Accordion.Collapse>
        </Card>
      </Accordion>
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setCategoryFilter: (text) => dispatch(setCategoryFilter(text)),
  setBusinessFilter: (text) => dispatch(setBusinessFilter(text)),
  setNoteFilter: (text) => dispatch(setNoteFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
