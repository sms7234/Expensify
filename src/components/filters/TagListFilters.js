import React from 'react';
import {connect} from 'react-redux';
import {setTagFilter} from '../../actions/filters';

export class TagListFilters extends React.Component {
  onTextChange = (e) => {
    this.props.setTagFilter(e.target.value);
  };
  render () {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="search tags"
              value={this.props.filters.tag}
              onChange={this.onTextChange}
            />
          </div>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTagFilter: (text) => dispatch(setTagFilter(text)),
});

export default connect(mapStateToProps, mapDispatchToProps)(TagListFilters);
