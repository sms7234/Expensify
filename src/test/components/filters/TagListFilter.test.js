import React from 'react';
import {shallow} from 'enzyme';
import {TagListFilters} from '../../../components/filters/TagListFilters';

let setTagFilter, wrapper;

const filters = {
  tag: ''
};

beforeEach(()=> {
  setTagFilter = jest.fn();
  wrapper=shallow(
    <TagListFilters
      filters={filters}
      setTagFilter={setTagFilter} />
  );
});

test('should render TagListFilters correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle text input', () => {
  const value="texxxxxt";
  wrapper.find('input').simulate('change', {
    target: {value}
  });
  expect(setTagFilter).toHaveBeenLastCalledWith(value);
});
