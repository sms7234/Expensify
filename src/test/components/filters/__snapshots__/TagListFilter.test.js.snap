// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TagListFilters correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="search tags"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
